client/beefy/src/worker.rs

```diff
@@ -460,24 +572,36 @@ where
 	header.digest().convert_first(|l| l.try_to(id).and_then(filter))
 }
 
-/// Calculate next block number to vote on
-fn vote_target<N>(best_grandpa: N, best_beefy: N, min_delta: u32) -> N
+/// Calculate next block number to vote on.
+fn vote_target<N>(best_grandpa: N, best_beefy: N, session_start: N, min_delta: u32) -> N
```

Comment:
```suggestion
fn vote_target<N>(best_grandpa: N, best_beefy: Option<N>, session_start: N, min_delta: u32) -> N
```

IMHO better to handle the missing `best_beefy` case in this function instead of unwrapping and assuming it will hit the first condition.